### PR TITLE
Milestone 5: refactor the codes

### PR DESCRIPTION
**Pull request by medaminedev66**
- Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Pay special attention to the "Single-responsiblity Principle".
- Create new classes: `HandleBooks, HandleRentals, HandlePeople, and Main`

### CODE REVIEW

File: handle_rentals.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    puts 'Id:'
+    id = gets.chomp
+    @handle_people.people.each { |person| @selected_person = person if person.id == id }
+    if @selected_person
+      puts "List of all rentals for #{@selected_person.name}, id: #{@selected_person.id}:"
+      @selected_person.rentals.each do |rental|
+        puts "- Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}\n"
+      end
+    else
+      puts "The member with id: #{id} doesn't exist"
```
Comment (Dipeshtwis) [inline handle_rentals.rb]: - [x] Looks like there is some issue here, If I am asking for the person whose id is `820`, it's giving me the rental detail for the person whose id is `716`. Please have a look into this.

Screenshot from 2021-11-15 19-05-54

File: handle_rentals.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    @handle_people = people
+    @handle_books = books
+  end
+
+  def create_rental
+    puts "Select book from this books using the number: \n"
+    @handle_books.list_of_books
+    book_number = gets.chomp
+    @handle_books.books.each_with_index { |book, index| @selected_book = book if (index + 1) == book_number.to_i }
+    puts "Select a person fome the following list of members buy id:\n"
```
Comment (Dipeshtwis) [inline handle_rentals.rb]: - [x] There is some typo here. Instead of `buy`, it should be `by` and instead of `fome`, it should be `from`. Please fix that

**Review by Dipeshtwis — CHANGES_REQUESTED** [review]
Hi @medaminedev66 and @julie-ify, 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


 ## Required Changes ♻️

**_Check the comments under the review_**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N.A.

Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**_Dipesh Kumar_**


------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project(check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: handle_rentals.rb (hunk 1 above)
Comment (medaminedev66) [reply handle_rentals.rb]: Ok. Thanks. We're going to solve this bug

File: handle_rentals.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+        puts "The member with id: #{id} doesn't have rentals"
+      else
+        puts "List of all rentals for #{@selected_person.name}, id: #{@selected_person.id}:"
+        @selected_person.rentals.each do |rental|
+          puts "- Date: #{rental.date}, Book: #{rental.book.title} by #{rental.book.author}\n"
+        end
+      end
+    end
+    @selected_person = 'null'
+  end
```
Comment (mikethreels) [inline handle_rentals.rb]: - [x] good job on changing the typo's notified by the previous CR 👍 however the issue when listing the rentals for a specific ID is not yet fixed, when looking for rentals from person ID 566 it provides me with the rentals of person ID 521, and when asking for the rentals for person ID 521 it says this person has no rentals
Przechwytywanie

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by medaminedev66** [comment]
We changed the algorithm of our method `list_rentals_by_id` and the app is working well when you try to list rentals by id.
To make sure that the app works well, we recorded a video that explains all the test processes: \
[Test our ruby app](https://www.youtube.com/watch?v=nU606mFC9G4)

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Now it is working! 👏 
Your project is complete! it's time to merge it 🚀 :
Congratulations! 🎉 

- [OPTIONAL] one thing that wasn't mentioned before and therefore it is an OPTIONAL suggestion, when creating a teacher it also asks if the person has permission, however, this is not required in case the person is a teacher.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by medaminedev66** [comment]
Thanks for the suggestion. We will work on it.

**Pull request merged**